File: src/main/scala/com/github/ai/split/entity/Expense.scala
package com.github.ai.split.entity

import java.util.UUID

case class Expense(
  uid: UUID,
  groupUid: UUID,
  title: String,
  description: String,
  amount: Double,
  paidBy: List[UUID],
  splitBetween: List[UUID]
)

File: android/app/src/main/java/com/github/ai/simplesplit/android/model/Expense.kt
package com.github.ai.simplesplit.android.model

import com.github.ai.simplesplit.android.data.database.model.CurrencyEntity
import kotlinx.serialization.Serializable

@Serializable
data class Expense(
    val uid: String,
    val title: String,
    val description: String,
    val amount: Double,
    val currency: CurrencyEntity,
    val paidBy: List<Member>,
    val splitBetween: List<Member>,
    val createdSeconds: Long,
    val modifiedSeconds: Long
)